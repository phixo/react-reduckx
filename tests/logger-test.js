/* eslint-disable no-console */
import Logger from '../src/logger';

describe('Reducks | logger', () => {
    const mockAction = {
        type: 'MOCK-ACTION',
    };

    const mockPrevState = {
        foo: 'bar',
    };

    const mockNextState = {
        foo: 'baz',
    };

    const spyGroup = jest.spyOn(console, 'group');
    const spyGroupEnd = jest.spyOn(console, 'groupEnd');
    const spyLog = jest.spyOn(console, 'log');

    beforeEach(() => {
        spyGroup.mockReset();
        spyGroupEnd.mockReset();
        spyLog.mockReset();
    });

    it('Should log the action to the console', () => {
        Logger.log(mockAction, mockPrevState, mockNextState);
        expect(spyGroup).toHaveBeenCalledWith(mockAction.type);
        expect(spyGroupEnd).toHaveBeenCalled();
        expect(spyLog).toHaveBeenCalled();

        expect(spyLog).toMatchSnapshot();
    });

    it(`Shouldn't log to the console when there's no action`, () => {
        Logger.log();
        expect(spyGroup).not.toHaveBeenCalled();
        expect(spyGroupEnd).not.toHaveBeenCalled();
        expect(spyLog).not.toHaveBeenCalled();
    });
});
