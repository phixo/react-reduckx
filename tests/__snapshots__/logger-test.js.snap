// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducks | logger Should log the action to the console 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "%cprev state",
      "color: #9E9E9E;",
      Object {
        "foo": "bar",
      },
    ],
    Array [
      "%caction    ",
      "color: #03A9F4;",
      Object {
        "type": "MOCK-ACTION",
      },
    ],
    Array [
      "%cnext state",
      "color: #4CAF50;",
      Object {
        "foo": "baz",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
